File: data structure/ll.cpp
#include<iostream>
using namespace std;

class node{
    public:
    int data;
    node* next;

    node(int val)
    {
        data=val;
        next=NULL;
    }
};

 void insertAtTail(node* &head, int val){
    node* n=new node(val);
    if(head==NULL)
    {
        head=n;
        return;
    }
    node* temp=head;
    while(temp->next!=NULL)
    {
        temp=temp->next;
    }
    temp->next=n;
    void display(node* head)
    {
        node* temp=head;
        while(temp!=NULL)
        {
            cout<<temp->data<<" ";
            temp=temp->next;
        }
        cout<<endl;
    }
 }

 bool search(node* head, int key)
 {
    node* temp=head;
    while(temp!=NULL)
    {
        if(temp->data==key)
        {
            return true;
        }
        return false;
    }
 }
 int main()
 {
    insertAtTail(head,1);
    insertAtTail(head,2);
    display(head):
    cout<<search(head,5)<<endl;
 }

File: java/LL.java
class LL{
    Node head;
    class Node{
        String data;
        node next;
    }

    Node(String data){
        this.data=data;
        this.next=null;
    }

    //add--first,last..
    public void addFirst(String data)
    {
        Node newNode=new Node(data);
        if(head==null)
        {
            head=newNode;
            return;
        }

        newNode.next=head;
        head=newNode;
    }

    public void addLast(String data)
    {
        Node newNode=new Node(data);
        if(head==null)
        {
            head=newNode;
            return;
        }

        Node currNde=head;
        while(currNode.next!=null)
        {
            currNode=currNode.next;
        }
        currNode.next=newNode;
    }

    //print
    public void printList(){
        if(head==null)
        {
            System.out.println("list is empty..");
            return;
        }

        Node currNode=head;
        while(currNode!=null)
        {
            System.out.print(currNode.data+"->");
            currNode=currNode.next;
        }
        System.out.println("null");
    }
}
public class LL {
    public static void main(String args[])
    {
        LL list=new LL();
        list.addFirst("a");
        list.addFirst("is");
        list.printList();
    }
    
}
